sphinx/tutorial/rose/furthertopics/rose-stem.rst

```diff
@@ -0,0 +1,309 @@
+execute ``rose stem`` on the command line.
+
+
+The ``rose_ana_position`` app
+-----------------------------
+
+The final component is a ``rose ana`` app to test the position of our
+spaceship matches the correct output.
+
+Create an app named ``rose_ana_position`` and paste the following into its
+``rose-app.conf`` file:
+
+.. code-block:: rose
+
+   [ana:grepper.FilePattern(Check X position at each timestep)]
+   pattern='^\s*Position:\s*(.*?)\s*,'
+   files=/home/user/spaceship/kgo.txt
+        =../spaceship/output.txt
+
+   [ana:grepper.FilePattern(Check Y position at each timestep)]
+   pattern='^\s*Position:.*?,\s*(.*?)\s*,'
+   files=/home/user/spaceship/kgo.txt
+        =../spaceship/output.txt
+
+   [ana:grepper.FilePattern(Check Z position at each timestep)]
+   pattern='^\s*Position:.*,\s*(.*)\s*$'
+   files=/home/user/spaceship/kgo.txt
+        =../spaceship/output.txt
+
+This will check that the positions reported by the program match those
+within the known good output file.
+
+
+Known Good Output
+-----------------
+
+In the root of the working copy is a file called ``kgo.txt``.
+
+Replace the ``/home/user/spaceship`` paths in the ``rose_ana_position``
+app with the path to this file.
+
+The known good output should be the result of a control run. ``rose ana``
+will compare the answers from this file (obtained using the extract and
+comparison methods in the ``rose-app.conf`` file) with the results from
+the user's code change.
+
+
+Adding the suite to version control
+-----------------------------------
+
+Before running the suite we need to make sure that all the files and
+directories we have created are known to the version control system.
+
+Add all the new files you've created using ``fcm add -c`` *(answer yes
+to the prompts)*.
+
+
+Running the test suite
+----------------------
+
+We should now be able to run the test suite. Simply type::
+
+   rose stem --group=command_spaceship
+
+anywhere in your working copy (the ``--source`` argument defaults to.
+so it should automatically pick up your working copy as the source).
+
+.. note::
+
+   If your site uses a cylc server, and your home directory is not shared
+   with the cylc server, you will need to add the option::
+
+      --host=localhost
+
+We use ``--group`` in preference to ``--task`` in this suite (both are
+synonymous) as we specify a group of tasks set up in the Jinja2 variable
+``name_graphs``.
+
+
+A failing test
+--------------
+
+Now edit the file::
+
+   rose-stem/app/spaceship/rose-app.conf
+
+and change one of the thrusts, then rerun ``rose stem``. You will find the
+``rose ana`` task fails, as the results have changed.
+
+Try modifying the Fortran source code - for example, changing the direction
+in which thrust is applied (by changing the acceleration to be subtracted
+from the velocity rather than added). Again, rerun ``rose stem``, and see
+the failure.
+
+In this way, you can monitor whether the behaviour of code is changed by
```

Comment:
Can we highlight this final sentence? It seems like it could be very useful as a general method for users to employ on their own code. Perhaps pop this in a '.. note::' box?